tests/integration/api/test_datetime.py

```diff
@@ -138,3 +144,70 @@ def test_to_timestamp_from_epoch(event_table, config, session):
         df_expected,
         dict_like_columns=["test_to_timestamp_from_epoch_feature"],
     )
+
+
+def test_date_difference_of_varchar_timestamp_with_timezone(event_table, config, session):
+    """
+    Test date difference of varchar timestamp with timezone
+    """
+    source_type = session.source_type
+    if source_type in SourceType.java_time_format_types():
+        format_string = "yyyy-MM-dd HH:mm:ss"
```

Comment:
I think it would be great if the test used very non-standard formats (ones that a plain `pd.to_datetime` would fail to parse) to ensure that the transformations don't rely on any implicit inference behavior.